src/raftstore/store/peer.rs

```diff
@@ -938,39 +989,31 @@ impl Peer {
         let mut ctx = ExecContext {
             snap: Snapshot::new(engine),
             apply_state: self.get_store().apply_state.clone(),
-            wb: WriteBatch::new(),
+            wb: wb,
             req: req,
         };
-        let (mut resp, exec_result) = self.exec_raft_cmd(&mut ctx).unwrap_or_else(|e| {
+        let (resp, exec_result) = self.exec_raft_cmd(&mut ctx).unwrap_or_else(|e| {
             error!("{} execute raft command err: {:?}", self.tag, e);
             (cmd_resp::new_error(e), None)
         });
 
         ctx.apply_state.set_applied_index(index);
         ctx.save(self.region_id).expect("save state must not fail");
```

Comment:
This key seems be put several times. Can it be only set when the WriteBatch is about to be written?